File: converted_source/usplash.lfm.py
from common import *

#Useage:
#        import SplashWindow
#        window = SplashWindow.SplashWindow()
#        window.visible = 1

class SplashWindow(JFrame):
    def __init__(self, title='FIX WINDOW TITLE'):
        JFrame.__init__(self, title, windowClosing=self.OnClose)
        self.setDefaultCloseOperation(WindowConstants.DISPOSE_ON_CLOSE) # HIDE_ON_CLOSE DO_NOTHING_ON_CLOSE
        self.BuildWindow()
        
    def OnClose(self, event):
        print "Closed"
        
    # a menu is top item on the menu bar and can be pulled down for a list of menu items
    def createMenu(self, name, mnemonic=None, menuBar = None):
        menu = JMenu(name)
        if mnemonic:
            menu.setMnemonic(mnemonic)
        if menuBar:
            menuBar.add(menu)
        return menu
        
    def BuildWindow(self):
        toolkit = java.awt.Toolkit.getDefaultToolkit()
        contentPane = self.getContentPane()
        contentPane.setLayout(None)
        # -- contentPane.setLayout(BoxLayout(contentPane, BoxLayout.Y_AXIS))
        buttonFont = Font("Arial Narrow", Font.PLAIN, 9)
        buttonMargin = Insets(1, 1, 1, 1)
        self.setBounds(349, 213, 379, 269  + 80) # extra for title bar and menu
        #  --------------- UNHANDLED ATTRIBUTE: self.BorderIcons = []
        #  --------------- UNHANDLED ATTRIBUTE: self.BorderStyle = bsNone
        #  --------------- UNHANDLED ATTRIBUTE: self.TextHeight = 14
        self.keyTyped = self.FormKeyPress
        self.keyPressed = self.FormKeyDown
        #  --------------- UNHANDLED ATTRIBUTE: self.OnCreate = FormCreate
        #  --------------- UNHANDLED ATTRIBUTE: self.OnActivate = FormActivate
        #  --------------- UNHANDLED ATTRIBUTE: self.KeyPreview = True
        #  --------------- UNHANDLED ATTRIBUTE: self.Position = poScreenCenter
        self.mousePressed = self.FormMouseDown
        #  --------------- UNHANDLED ATTRIBUTE: self.Scaled = False
        #  --------------- UNHANDLED ATTRIBUTE: self.PixelsPerInch = 96
        self.mouseClicked = self.FormClick
        
        self.splashImage256ColorsImage = toolkit.createImage("../resources/SplashForm_splashImage256Colors.png")
        self.splashImage256Colors = ImagePanel(ImageIcon(self.splashImage256ColorsImage))
        self.splashImage256Colors.setBounds(0, 0, 382, 270)
        self.splashImage256Colors.visible = 0
        self.splashImage256Colors.mouseClicked = self.splashImage256ColorsClick
        self.versionLabel = JLabel('version 0.1', font=buttonFont)
        self.versionLabel.setBounds(13, 81, 68, 16)
        self.versionLabel.mouseClicked = self.controlClick
        self.loadingLabel = JLabel('Registered to:', font=buttonFont)
        self.loadingLabel.setBounds(33, 107, 88, 16)
        self.loadingLabel.mouseClicked = self.controlClick
        self.codeLabel = JLabel('Name', font=buttonFont)
        self.codeLabel.setBounds(127, 107, 37, 16)
        self.codeLabel.visible = 0
        self.codeLabel.mouseClicked = self.controlClick
        self.splashImageTrueColorImage = toolkit.createImage("../resources/SplashForm_splashImageTrueColor.png")
        self.splashImageTrueColor = ImagePanel(ImageIcon(self.splashImageTrueColorImage))
        self.splashImageTrueColor.setBounds(109, -167, 382, 270)
        self.splashImageTrueColor.visible = 0
        self.splashImageTrueColor.mouseClicked = self.splashImage256ColorsClick
        self.closeImage = toolkit.createImage("../resources/SplashForm_close.png")
        self.close = SpeedButton(ImageIcon(self.closeImage), actionPerformed=self.closeClick, font=buttonFont, margin=buttonMargin)
        self.close.setBounds(274, 134, 92, 29)
        #  --------------- UNHANDLED ATTRIBUTE: self.close.Flat = True
        self.close.visible = 0
        self.supportButtonImage = toolkit.createImage("../resources/SplashForm_supportButton.png")
        self.supportButton = SpeedButton(ImageIcon(self.supportButtonImage), actionPerformed=self.supportButtonClick, font=buttonFont, margin=buttonMargin)
        self.supportButton.setBounds(274, 171, 92, 26)
        #  --------------- UNHANDLED ATTRIBUTE: self.supportButton.Flat = True
        self.supportButton.visible = 0
        contentPane.add(self.splashImage256Colors)
        contentPane.add(self.versionLabel)
        contentPane.add(self.loadingLabel)
        contentPane.add(self.codeLabel)
        contentPane.add(self.splashImageTrueColor)
        contentPane.add(self.close)
        contentPane.add(self.supportButton)
        
    def FormClick(self, event):
        print "FormClick"
        
    def FormKeyDown(self, event):
        print "FormKeyDown"
        
    def FormKeyPress(self, event):
        print "FormKeyPress"
        
    def FormMouseDown(self, event):
        print "FormMouseDown"
        
    def closeClick(self, event):
        print "closeClick"
        
    def controlClick(self, event):
        print "controlClick"
        
    def splashImage256ColorsClick(self, event):
        print "splashImage256ColorsClick"
        
    def supportButtonClick(self, event):
        print "supportButtonClick"
        
if __name__ == "__main__":
    window = SplashWindow()
    window.visible = 1
    window.setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE)

File: converted_source/usplash.lfm.java

import java.awt.*;
import java.awt.event.*;
import javax.swing.*;
import javax.swing.event.*;
import javax.swing.table.*;
// import common.*;

public class SplashWindow extends JFrame {
    private static final long serialVersionUID = 1L;
    ImagePanel splashImage256Colors;
    JLabel versionLabel;
    JLabel loadingLabel;
    JLabel codeLabel;
    ImagePanel splashImageTrueColor;
    SpeedButton close;
    SpeedButton supportButton;
    public JPanel mainContentPane;
    public JPanel delphiPanel;
    
    public SplashWindow() {
        super();
        this.setDefaultCloseOperation(WindowConstants.DISPOSE_ON_CLOSE); // HIDE_ON_CLOSE DO_NOTHING_ON_CLOSE
        initialize();
    }
    
    private void initialize() {
        this.setSize(new Dimension(705, 325));
        this.setContentPane(getMainContentPane());
        this.setTitle("FIX WINDOW TITLE");
    }
    
    private JPanel getMainContentPane() {
        if (mainContentPane == null) {
            mainContentPane = new JPanel();
            mainContentPane.setLayout(new BorderLayout());
            mainContentPane.add(getDelphiPanel(), BorderLayout.CENTER);
        }
        return mainContentPane;
    }
    public void OnClose(ActionEvent event) {
        System.out.println("Closed");
    }
    
    public JPanel getDelphiPanel() {
        if (delphiPanel != null) return delphiPanel;
        Toolkit toolkit = java.awt.Toolkit.getDefaultToolkit();
        delphiPanel = new JPanel();
        delphiPanel.setLayout(null);
        // -- delphiPanel.setLayout(BoxLayout(contentPane, BoxLayout.Y_AXIS));
        // Font buttonFont = new Font("Arial Narrow", Font.PLAIN, 9);
        // Insets buttonMargin = new Insets(1, 1, 1, 1);
        this.setBounds(349, 213, 379, 269  + 80); // extra for title bar and menu
        //  --------------- UNHANDLED ATTRIBUTE: this.BorderIcons = [];
        //  --------------- UNHANDLED ATTRIBUTE: this.BorderStyle = bsNone;
        //  --------------- UNHANDLED ATTRIBUTE: this.TextHeight = 14;
        this.addKeyListener(new java.awt.event.KeyAdapter() {
            public void keyTyped(java.awt.event.KeyEvent event) {
                FormKeyPress(event);
            }
        });
        this.addKeyListener(new java.awt.event.KeyAdapter() {
            public void keyPressed(java.awt.event.KeyEvent event) {
                FormKeyDown(event);
            }
        });
        //  --------------- UNHANDLED ATTRIBUTE: this.OnCreate = FormCreate;
        //  --------------- UNHANDLED ATTRIBUTE: this.OnActivate = FormActivate;
        //  --------------- UNHANDLED ATTRIBUTE: this.KeyPreview = True;
        //  --------------- UNHANDLED ATTRIBUTE: this.Position = poScreenCenter;
        this.addMouseListener(new java.awt.event.MouseAdapter() {
            public void mousePressed(java.awt.event.MouseEvent event) {
                FormMouseDown(event);
            }
        });
        //  --------------- UNHANDLED ATTRIBUTE: this.Scaled = False;
        //  --------------- UNHANDLED ATTRIBUTE: this.PixelsPerInch = 96;
        this.addMouseListener(new java.awt.event.MouseAdapter() {
            public void mouseClicked(java.awt.event.MouseEvent event) {
                FormClick(event);
            }
        });
        
        
        Image splashImage256ColorsImage = toolkit.createImage("../resources/SplashForm_splashImage256Colors.png");
        this.splashImage256Colors = new ImagePanel(new ImageIcon(splashImage256ColorsImage));
        this.splashImage256Colors.setBounds(0, 0, 382, 270);
        this.splashImage256Colors.setVisible(false);
        this.splashImage256Colors.addMouseListener(new java.awt.event.MouseAdapter() {
            public void mouseClicked(java.awt.event.MouseEvent event) {
                splashImage256ColorsClick(event);
            }
        });
        
        this.versionLabel = new JLabel("version 0.1");
        this.versionLabel.setBounds(13, 81, 68, 16);
        this.versionLabel.addMouseListener(new java.awt.event.MouseAdapter() {
            public void mouseClicked(java.awt.event.MouseEvent event) {
                controlClick(event);
            }
        });
        
        this.loadingLabel = new JLabel("Registered to:");
        this.loadingLabel.setBounds(33, 107, 88, 16);
        this.loadingLabel.addMouseListener(new java.awt.event.MouseAdapter() {
            public void mouseClicked(java.awt.event.MouseEvent event) {
                controlClick(event);
            }
        });
        
        this.codeLabel = new JLabel("Name");
        this.codeLabel.setBounds(127, 107, 37, 16);
        this.codeLabel.setVisible(false);
        this.codeLabel.addMouseListener(new java.awt.event.MouseAdapter() {
            public void mouseClicked(java.awt.event.MouseEvent event) {
                controlClick(event);
            }
        });
        
        Image splashImageTrueColorImage = toolkit.createImage("../resources/SplashForm_splashImageTrueColor.png");
        this.splashImageTrueColor = new ImagePanel(new ImageIcon(splashImageTrueColorImage));
        this.splashImageTrueColor.setBounds(109, -167, 382, 270);
        this.splashImageTrueColor.setVisible(false);
        this.splashImageTrueColor.addMouseListener(new java.awt.event.MouseAdapter() {
            public void mouseClicked(java.awt.event.MouseEvent event) {
                splashImage256ColorsClick(event);
            }
        });
        
        Image closeImage = toolkit.createImage("../resources/SplashForm_close.png");
        this.close = new SpeedButton(new ImageIcon(closeImage));
        this.close.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent event) {
                closeClick(event);
            }
        });
        this.close.setBounds(274, 134, 92, 29);
        //  --------------- UNHANDLED ATTRIBUTE: this.close.Flat = True;
        this.close.setVisible(false);
        
        Image supportButtonImage = toolkit.createImage("../resources/SplashForm_supportButton.png");
        this.supportButton = new SpeedButton(new ImageIcon(supportButtonImage));
        this.supportButton.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent event) {
                supportButtonClick(event);
            }
        });
        this.supportButton.setBounds(274, 171, 92, 26);
        //  --------------- UNHANDLED ATTRIBUTE: this.supportButton.Flat = True;
        this.supportButton.setVisible(false);
        
        delphiPanel.add(splashImage256Colors);
        delphiPanel.add(versionLabel);
        delphiPanel.add(loadingLabel);
        delphiPanel.add(codeLabel);
        delphiPanel.add(splashImageTrueColor);
        delphiPanel.add(close);
        delphiPanel.add(supportButton);
        return delphiPanel;
    }
    
        
    public void FormClick(MouseEvent event) {
        System.out.println("FormClick");
    }
        
    public void FormKeyDown(KeyEvent event) {
        System.out.println("FormKeyDown");
    }
        
    public void FormKeyPress(KeyEvent event) {
        System.out.println("FormKeyPress");
    }
        
    public void FormMouseDown(MouseEvent event) {
        System.out.println("FormMouseDown");
    }
        
    public void closeClick(ActionEvent event) {
        System.out.println("closeClick");
    }
        
    public void controlClick(MouseEvent event) {
        System.out.println("controlClick");
    }
        
    public void splashImage256ColorsClick(MouseEvent event) {
        System.out.println("splashImage256ColorsClick");
    }
        
    public void supportButtonClick(ActionEvent event) {
        System.out.println("supportButtonClick");
    }

    public static void main(String[] args) {
        SwingUtilities.invokeLater(new Runnable() {
            public void run() {
                SplashWindow thisClass = new SplashWindow();
                thisClass.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
                thisClass.setVisible(true);
            }
        });
    }

}
